packages-exp/auth-exp/src/core/user/user_impl.ts

```diff
@@ -185,7 +193,7 @@ export class UserImpl implements User {
 
     assert(uid && plainObjectTokenManager, auth.name);
 
-    const stsTokenManager = StsTokenManager.fromPlainObject(
+    const stsTokenManager = StsTokenManager.fromJSON(
```

Comment:
You added several bits to `toJSON` that need to be carried into this:

providerData,
tenantID, for example

Reply:
done, also removed the auth properties like apiKey since we don't need them on user now due to User deferring to its parent Auth in the new code